src/sentry/api/endpoints/project_app_store_connect_credentials.py

```diff
@@ -265,6 +302,10 @@ def post(self, request: Request, project: Project) -> Response:
             }
         )
 
+        redacted_config = validated_config.to_redacted_json()
+        for secret in get_secret_fields(validated_config.type):
+            config[secret] = redacted_config[secret]
+
         return Response(config, status=200)
```

Comment:
why do we need to iterate over all the fields here? that should be the job of `to_redacted_json`

I think you should just be able to straight up return the `redacted_config`.

Reply:
initially i was going to say that this was because the structure returned by this endpoint appears to be a subset of the actual config (see: no `appName`, no `appId`, etc but also the renaming of `orgPublicId)

digging into the frontend code a bit, it looks like we no longer use the return value of this endpoint. (see: https://github.com/getsentry/sentry/pull/28192/files#diff-e5ffd1c8a64b5ad8dbdffc5e8f95b1c1db09d91b1536ecfb53fbb2f21f981508R251)

we could probably just axe this response value and return the `id` or something 🤔

Reply:
slimmed down the response to just an ID and removed block of logic entirely. tested adding a new app store connect config as well as updating an existing one and it doesn't look like this breaks anything, so i'll be going with this.